lib/lightning/accounts/user.ex

```diff
@@ -164,14 +164,74 @@ defmodule Lightning.Accounts.User do
     |> put_change(:role, :superuser)
   end
 
-  def validate_email(changeset) do
+  @doc """
+  Validates that a user has access to all projects associated with a credential.
+  Returns a changeset with any project access errors.
+  """
+  def validate_project_access(changeset, credential) do
+    case get_field(changeset, :email) do
+      nil ->
+        changeset
+
+      email ->
+        case Lightning.Repo.exists?(User |> where(email: ^email)) do
+          false ->
+            changeset
+
+          true ->
+            user = Lightning.Accounts.get_user_by_email(email)
+
+            projects =
+              credential
+              |> Lightning.Credentials.diff_project_credentials_and_project_users(
```

Comment:
What you think about naming this one by the purpose (goal of the function)? I was trying to understand this function because project credentials and project users are different entities so I was trying to grasp what would be the meaning to diff them? By reading your docs I could see that it's about detecting invalid projects.

Reply:
This is also done, thanks a lot